x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DestinationIndex.java

```diff
@@ -303,6 +308,18 @@ public static void updateMappingsToDestIndex(Client client,
         getFieldCapsForRequiredFields(client, config, fieldCapabilitiesListener);
     }
 
+    private static ImmutableOpenMap<String, MappingMetadata> createEmptyMappings() {
+        MappingMetadata singleDocEmptyMetadata;
+        try {
+            singleDocEmptyMetadata = new MappingMetadata(MapperService.SINGLE_MAPPING_NAME, Collections.emptyMap());
+        } catch (IOException e) {
+            throw new IllegalStateException("we should never fail to create empty mapping_metadata");
+        }
+        return ImmutableOpenMap.<String, MappingMetadata>builder()
+            .fPut(MapperService.SINGLE_MAPPING_NAME, singleDocEmptyMetadata)
```

Comment:
`MapperService.SINGLE_MAPPING_NAME` and inside

`singleDocEmptyMetadata = new MappingMetadata(MapperService.SINGLE_MAPPING_NAME, Collections.emptyMap());` 

???

Reply:
Yes, it looks weird but this is how it's structured in `7.x`. The map contains the type (`_doc`) as a key and the value is a `MappingMetadata` object which also has the type.